authors/apps/report/models.py

```diff
@@ -0,0 +1,16 @@
+from django.db import models
+from authors import settings
+from authors.apps.articles.models import Article
+from authors.apps.profiles.models import Profile
+
+# Create your models here.
+class ReportArticle(models.Model):
+    """model for reporting an article"""
+    reporter = models.ForeignKey(Profile, on_delete=models.CASCADE)
+    article = models.ForeignKey(Article, to_field="slug", on_delete=models.CASCADE)
+    violation_subject = models.CharField(max_length=100, blank=False, null=False)
```

Comment:
well done @llwasampijja, I was able to report a violation and also noticed that I was not able to report my own article which is good. However, I was thinking of the violation_subject field as a ChoiceField like how it is on medium.com where the user selects the category of the violation for which they are reporting the article. We predefine the categories for violations.